lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2443,9 +2443,9 @@ def bar3d(self, x, y, z, dx, dy, dz, color=None,
 
         facecolors = []
         if color is None:
-            # no color specified
-            facecolors = [None] * len(x)
-        elif len(color) == len(x):
+            color = self._get_lines.get_next_color()
```

Comment:
This is a string, generally, no? Which means it could match the next condition if there are 6 bars, or 1 bar, or whatever the length of the colour string is. I _think_ that maybe isn't expected to happen?